mqtt/mqtt-broker/src/auth/authorization.rs

```diff
@@ -0,0 +1,217 @@
+#![allow(dead_code)]
+use async_trait::async_trait;
+use mqtt3::proto;
+
+use crate::{AuthId, ClientId, Error};
+
+/// A trait to check a MQTT client permissions to perform some actions.
+#[async_trait]
+pub trait Authorizer {
+    /// Authorizes a MQTT client to perform some action.
+    async fn authorize(&self, activity: Activity) -> Result<bool, Error>;
+}
+
+#[async_trait]
+impl<F> Authorizer for F
+where
+    F: Fn(Activity) -> Result<bool, Error> + Sync,
+{
+    async fn authorize(&self, activity: Activity) -> Result<bool, Error> {
+        self(activity)
+    }
+}
+
+/// Default implementation that always denies any operation a client intends to perform.
+/// This implementation will be used if custom authorization mechanism was not provided.
+pub struct DefaultAuthorizer;
+
+#[async_trait]
+impl Authorizer for DefaultAuthorizer {
+    async fn authorize(&self, _: Activity) -> Result<bool, Error> {
+        Ok(false)
```

Comment:
Should the default authorize allow everything rather than deny? Presumably the default authorizer will only be used for tests, never in production?

Reply:
Default authorizer will deny every request unless configured other way. In production code when we configure broker we set another authorizer that allows everything for now. Once we have edgehub integration we will set a new edgehub-authorizer instead.